tests/wallet/nft_wallet/test_nft_off_chain.py

```diff
@@ -0,0 +1,116 @@
+from typing import Tuple
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import Program
+from chia.util.ints import uint32
+from chia.wallet.nft_wallet.nft_info import NFTCoinInfo
+from chia.wallet.nft_wallet.nft_off_chain import (
+    delete_off_chain_metadata,
+    get_off_chain_metadata,
+    read_off_chain_metadata,
+)
+from chia.wallet.nft_wallet.nft_puzzles import (
+    LAUNCHER_PUZZLE_HASH,
+    NFT_TRANSFER_PROGRAM_DEFAULT,
+    SINGLETON_MOD_HASH,
+    construct_ownership_layer,
+    create_full_puzzle,
+)
+from chia.wallet.puzzles.p2_delegated_puzzle_or_hidden_puzzle import puzzle_for_pk, solution_for_conditions
+from tests.core.make_block_generator import int_to_public_key
+from tests.wallet.nft_wallet.test_nft_puzzles import NFT_METADATA_UPDATER_DEFAULT, OFFER_MOD
+
+
+def make_a_new_solution() -> Tuple[Program, Program]:
+    destination = int_to_public_key(2)
+    p2_puzzle = puzzle_for_pk(destination)
+    puzhash = p2_puzzle.get_tree_hash()
+    new_did = Program.to("test").get_tree_hash()
+    new_did_inner_hash = Program.to("fake").get_tree_hash()
+    trade_prices_list = [[200, OFFER_MOD.get_tree_hash()]]
+
+    condition_list = [
+        [
+            51,
+            puzhash,
+            1,
+            [puzhash],
+        ],
+        [-10, new_did, trade_prices_list, new_did_inner_hash],
+    ]
+    solution = Program.to(
+        [
+            [],
+            [],
+            [
+                [solution_for_conditions(condition_list)],
+            ],
+        ],
+    )
+    return p2_puzzle, solution
+
+
+def make_a_new_ownership_layer_puzzle() -> Tuple[Program, Program]:
+    pubkey = int_to_public_key(1)
+    innerpuz = puzzle_for_pk(pubkey)
+    old_did = Program.to("test_2").get_tree_hash()
+    nft_id = Program.to("nft_id")
+    SINGLETON_STRUCT = Program.to((SINGLETON_MOD_HASH, (nft_id, LAUNCHER_PUZZLE_HASH)))
+    curried_tp = NFT_TRANSFER_PROGRAM_DEFAULT.curry(
+        SINGLETON_STRUCT,
+        innerpuz.get_tree_hash(),
+        2000,
+    )
+    curried_inner = innerpuz
+    curried_ownership_layer = construct_ownership_layer(old_did, curried_tp, curried_inner)
+    return innerpuz, curried_ownership_layer
+
+
+def test_create_full_puzzle() -> Program:
```

Comment:
there are no asserts in this function, why is it prefixed with `test_`? looks like a helper function, so makes more sense to name it `create_test_full_puzzle`